main.go

```diff
@@ -120,41 +150,54 @@ func main() {
 		namespaceList = nsSlice(watchNamespaces)
 		logger.Info("watching namespaces: " + watchNamespaces)
 		mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-			Scheme:             scheme,
-			MetricsBindAddress: metricsAddr,
-			LeaderElection:     enableLeaderElection,
-			NewCache:           cache.MultiNamespacedCacheBuilder(namespaceList),
+			Scheme:                 scheme,
+			MetricsBindAddress:     metricsAddr,
+			HealthProbeBindAddress: probeAddr,
+			LeaderElection:         enableLeaderElection,
+			LeaderElectionID:       "5ac9a181.secrets-manager.tuenti.io",
+			NewCache:               cache.MultiNamespacedCacheBuilder(namespaceList),
 		})
 		if err != nil {
 			setupLog.Error(err, "unable to start manager")
 			os.Exit(1)
 		}
 	} else {
 		mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-			Scheme:             scheme,
-			MetricsBindAddress: metricsAddr,
-			LeaderElection:     enableLeaderElection,
+			Scheme:                 scheme,
+			MetricsBindAddress:     metricsAddr,
+			HealthProbeBindAddress: probeAddr,
+			LeaderElection:         enableLeaderElection,
+			LeaderElectionID:       "5ac9a181.secrets-manager.tuenti.io",
 		})
 		if err != nil {
 			setupLog.Error(err, "unable to start manager")
 			os.Exit(1)
 		}
 	}
 
-	err = (&controllers.SecretDefinitionReconciler{
+	if err = (&controllers.SecretDefinitionReconciler{
 		Backend:              *backendClient,
 		Client:               mgr.GetClient(),
+		Scheme:               mgr.GetScheme(),
```

Comment:
Not needed anymore as the context is passed to the `Reconcile` function.

Reply:
Removed